src/module_ncio.f90

```diff
@@ -840,42 +815,44 @@ end subroutine close_dataset
   !> subroutine read_vardata(dset,varname,values,nslice,slicedim,errcode)
   !! read data from variable varname in dataset dset, return in it array values.
   !!
-  !! dset:    Input dataset instance returned by open_dataset/create_dataset.
-  !! varname: Input string name of variable.
-  !! values:  Array to hold variable data.  Must be
+  !! @param[in] dset:    Input dataset instance returned by open_dataset/create_dataset.
+  !! @param[in] varname: Input string name of variable.
+  !! @param values:  Array to hold variable data.  Must be
   !!          an allocatable array with same rank
   !!          as variable varname (or 1 dimension less).
-  !! nslice:  optional index along dimension slicedim
-  !! slicedim: optional, if nslice is set, index of which dimension to slice with
-  !!          nslice, default is ndims
-  !! ncstart: optional, if ncstart and nccount are set, manually specify the
+  !! @param nslice:  optional index along dimension slicedim
```

Comment:
In the case of params, remove the ":" or "-" or whatever is used in the original docs to separate the name of the parameter from its description. In doxygen, just use a single space. Like this:

`!! @param nslice optional index along dimension slicedim
`